src/components/navigation-section.tsx

```diff
@@ -1,37 +1,35 @@
 import React from 'react';
-import { NavigationItemList, NavigationSectionItem } from '../types';
+import { NavigationSectionItem } from '../types';
 import NavigationItem from './navigation-item';
 
 type Props = {
   key: string;
   title: string;
-  section: NavigationItemList;
+  section: NavigationSectionItem[];
   currentSlug: string;
   onItemClick: () => void;
-  flatSections: NavigationSectionItem[];
+  readStatus: Map<NavigationSectionItem['slug'], boolean>;
```

Comment:
Curious question, instead of Map, could a Set be alternative, if the data structure just tracks the unique slugs, instead of tracking k-v pairs of (slugs, is done flags)?

Reply:
Hmm... that's an interesting proposition. 👍 

We could keep track of the unique slugs that have been marked as read in a set. And when rendering the navigation item, we could just check if the corresponding slug is present in the set and decide is it is read.

Seems like a good idea! Will make the data structure even leaner. Trying it out right away. 💯

Reply:
Done. And done. Also, threw in early returns from the iterations as soon as the current slug is met, for good measure.

It works great and is lighter! 💯 

Thanks for the suggestion, @BeniCheni! ❤️